frontend/javascripts/viewer/view/action-bar/starting_job_modals.tsx

```diff
@@ -712,28 +726,64 @@ function useCurrentlySelectedBoundingBox(
   return currentlySelectedBoundingBox;
 }
 
+// This function mirrors the selection of the mag
+// in voxelytics/worker/job_utils/voxelytics_utils.py select_mag_for_model_prediction
+const getBestFittingMagComparedToTrainingDS = (
+  colorLayer: APIDataLayer,
+  datasetScaleMag1: VoxelSize,
+  jobType: APIJobType.INFER_MITOCHONDRIA | APIJobType.INFER_NEURONS | APIJobType.INFER_NUCLEI,
+) => {
+  if (jobType === APIJobType.INFER_MITOCHONDRIA) {
+    // infer_mitochondria_model always infers on the finest mag of the current dataset
+    const magInfo = getMagInfo(colorLayer.resolutions);
+    return magInfo.getFinestMag();
+  }
+  const modelScale = MEAN_VX_SIZE[jobType];
+  let closestMagOfCurrentDS = colorLayer.resolutions[0];
+  let bestDifference = [
+    Number.POSITIVE_INFINITY,
+    Number.POSITIVE_INFINITY,
+    Number.POSITIVE_INFINITY,
+  ];
+
+  const datasetScaleInNm = convertVoxelSizeToUnit(datasetScaleMag1);
+
+  for (const mag of colorLayer.resolutions) {
+    const diff = datasetScaleInNm.map(
+      (dim, i) => Math.log2(dim * mag[i]) - Math.log2(modelScale[i]),
```

Comment:
Pythons and numpys log functions use the natural logarithm.

```suggestion
      (dim, i) => Math.log(dim * mag[i]) - Math.log(modelScale[i]),
```